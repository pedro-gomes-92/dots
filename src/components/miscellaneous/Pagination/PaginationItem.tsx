import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { TextButton, TextButtonProps } from '../../actions';

export interface PaginationItemProps extends TextButtonProps {
  text: string;
  isCurrent?: boolean;
}

export const PaginationItem: FunctionComponent<PaginationItemProps> = ({
  className,
  isCurrent,
  ...rest
}: PaginationItemProps): JSX.Element => (
  <TextButton className={classnames('pagination-link', { 'is-current': isCurrent }, className)} {...rest} />
);

PaginationItem.defaultProps = {
  isCurrent: false,
};
