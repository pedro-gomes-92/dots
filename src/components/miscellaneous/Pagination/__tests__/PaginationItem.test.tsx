import React from 'react';
import { render } from '@testing-library/react';

import { PaginationItem } from '../PaginationItem';

describe('Given <PaginationItem />', (): void => {
  let container: HTMLElement;

  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      container = render(<PaginationItem text="1" onClick={(): void => {}} />).container;
      expect(container).toMatchSnapshot();
    });

    describe('And uses text', (): void => {
      it('Then should render text value', (): void => {
        container = render(<PaginationItem text="2" onClick={(): void => {}} />).container;
        expect(container.querySelector('.text').textContent).toEqual('2');
      });
    });

    describe('And uses isCurrent', (): void => {
      it('Then should mark the item as active', (): void => {
        container = render(<PaginationItem text="2" isCurrent onClick={(): void => {}} />).container;
        expect(container.querySelector('.is-current')).toBeTruthy();
      });
    });
  });
});
