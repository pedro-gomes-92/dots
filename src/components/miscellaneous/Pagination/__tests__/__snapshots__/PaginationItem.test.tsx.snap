// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <PaginationItem /> When initializes Then should match the snapshot 1`] = `
<div>
  <button
    class="button is-text pagination-link"
  >
    <span
      class="text"
    >
      1
    </span>
  </button>
</div>
`;
